cmd/minikube/cmd/completion.go

```diff
@@ -106,175 +111,29 @@ func GenerateBashCompletion(w io.Writer, cmd *cobra.Command) error {
-		fi
-	fi
-	type "$@"
-}
-__minikube_compgen() {
-	local completions w
-	completions=( $(compgen "$@") ) || return $?
-	# filter by given word as prefix
-	while [[ "$1" = -* && "$1" != -- ]]; do
-		shift
-		shift
-	done
-	if [[ "$1" == -- ]]; then
-		shift
-	fi
-	for w in "${completions[@]}"; do
-		if [[ "${w}" = "$1"* ]]; then
-			echo "${w}"
-		fi
-	done
-}
-__minikube_compopt() {
-	true # don't do anything. Not supported by bashcompinit in zsh
-}
-__minikube_declare() {
-	if [ "$1" == "-F" ]; then
-		whence -w "$@"
-	else
-		builtin declare "$@"
-	fi
-}
-__minikube_ltrim_colon_completions()
-{
-	if [[ "$1" == *:* && "$COMP_WORDBREAKS" == *:* ]]; then
-		# Remove colon-word prefix from COMPREPLY items
-		local colon_word=${1%${1##*:}}
-		local i=${#COMPREPLY[*]}
-		while [[ $((--i)) -ge 0 ]]; do
-			COMPREPLY[$i]=${COMPREPLY[$i]#"$colon_word"}
-		done
-	fi
-}
-__minikube_get_comp_words_by_ref() {
-	cur="${COMP_WORDS[COMP_CWORD]}"
-	prev="${COMP_WORDS[${COMP_CWORD}-1]}"
-	words=("${COMP_WORDS[@]}")
-	cword=("${COMP_CWORD[@]}")
-}
-__minikube_filedir() {
-	local RET OLD_IFS w qw
-	__debug "_filedir $@ cur=$cur"
-	if [[ "$1" = \~* ]]; then
-		# somehow does not work. Maybe, zsh does not call this at all
-		eval echo "$1"
-		return 0
-	fi
-	OLD_IFS="$IFS"
-	IFS=$'\n'
-	if [ "$1" = "-d" ]; then
-		shift
-		RET=( $(compgen -d) )
-	else
-		RET=( $(compgen -f) )
-	fi
-	IFS="$OLD_IFS"
-	IFS="," __debug "RET=${RET[@]} len=${#RET[@]}"
-	for w in ${RET[@]}; do
-		if [[ ! "${w}" = "${cur}"* ]]; then
-			continue
-		fi
-		if eval "[[ \"\${w}\" = *.$1 || -d \"\${w}\" ]]"; then
-			qw="$(__minikube_quote "${w}")"
-			if [ -d "${w}" ]; then
-				COMPREPLY+=("${qw}/")
-			else
-				COMPREPLY+=("${qw}")
-			fi
-		fi
-	done
-}
-__minikube_quote() {
-	if [[ $1 == \'* || $1 == \"* ]]; then
-		# Leave out first character
-		printf %q "${1:1}"
-	else
-		printf %q "$1"
-	fi
-}
-autoload -U +X bashcompinit && bashcompinit
-# use word boundary patterns for BSD or GNU sed
-LWORD='[[:<:]]'
-RWORD='[[:>:]]'
-if sed --help 2>&1 | grep -q GNU; then
-	LWORD='\<'
-	RWORD='\>'
-fi
-__minikube_convert_bash_to_zsh() {
-	sed \
-	-e 's/declare -F/whence -w/' \
-	-e 's/_get_comp_words_by_ref "\$@"/_get_comp_words_by_ref "\$*"/' \
-	-e 's/local \([a-zA-Z0-9_]*\)=/local \1; \1=/' \
-	-e 's/flags+=("\(--.*\)=")/flags+=("\1"); two_word_flags+=("\1")/' \
-	-e 's/must_have_one_flag+=("\(--.*\)=")/must_have_one_flag+=("\1")/' \
-	-e "s/${LWORD}_filedir${RWORD}/__minikube_filedir/g" \
-	-e "s/${LWORD}_get_comp_words_by_ref${RWORD}/__minikube_get_comp_words_by_ref/g" \
-	-e "s/${LWORD}__ltrim_colon_completions${RWORD}/__minikube_ltrim_colon_completions/g" \
-	-e "s/${LWORD}compgen${RWORD}/__minikube_compgen/g" \
-	-e "s/${LWORD}compopt${RWORD}/__minikube_compopt/g" \
-	-e "s/${LWORD}declare${RWORD}/__minikube_declare/g" \
-	-e "s/\\\$(type${RWORD}/\$(__minikube_type/g" \
-	<<'BASH_COMPLETION_EOF'
-`
-
-	_, err := out.Write([]byte(zshAutoloadTag))
+	_, err := out.Write([]byte(boilerPlate))
```

Comment:
I also take this chance to simplify zsh completion code by using upstream library from cobra, instead of converting from bash to zsh. Please also let me know if it's fine, otherwise I will revert this change.

Reply:
Sounds good do you mind sharing before and after this PR the output I so t have a fish terminal myself can't verify it

Reply:
sure, let me add the terminal recording shortly

Reply:
@medyagh done, I have added the terminal recording for zsh and fish as well.

Reply:
oh new zsh completion was failing for sub command, let me revert changes for zsh for now.